corpus/declarations.txt

```diff
@@ -265,8 +265,10 @@ func f[A int|string, B ~int, C ~int|~string]()
         (identifier)
         (type_identifier))
       (parameter_declaration
-        (identifier)
-        (identifier)
+        (type_identifier))
+      (parameter_declaration
+        (type_identifier))
+      (parameter_declaration
```

Comment:
Unfortunately, it looks like we can't reuse parameter_list for receiver, parameters and result.

Basically, the receiver cannot take a list for parameters, it's always a single parameter, but `(x)` means that x is a `type_identifier`, whereas `(x y)` means that x is `identifier` and y is `type_identifier`.

Now, for parameters vs result:

- `(x int, y int)` means the same thing in both cases (x and y are `identifier`, both `int`s are `type_identifier`)
- `(x, y int)` means the same thing in both cases (x and y are `identifier`, int is a `type_identifier`)
- `(x, y)` is invalid for parameters, but in the case result x and y are `type_identifier`

I can try to send a PR later this week.

Reply:
waiting for your PR

Reply:
Hmm I'm actually struggling with this one. I wanted to make something like:

```go
func f1(x1, x2 int) {}
func f2(int, int) {}
```

Become:

```scheme
(function_declaration
  (identifier)
  (parameter_list
    (parameter_declaration
      (identifier))
    (parameter_declaration
      (identifier)
      (type_identifier)))
  (block))

(function_declaration
  (identifier)
  (parameter_list
    (type_identifier)
    (type_identifier))
  (block))
```

But that's surprisingly tricky 😁 

I tried:

```js
    parameter_list: $ => seq(
      '(',
      optional(
        seq(
          choice(
            commaSep1(choice($.parameter_declaration, $.variadic_parameter_declaration)),
            seq(commaSep($._type)),
          ),
          optional(','),
        ),
      ),
      ')'
    ),

    parameter_declaration: $ => prec.left(-1, seq(
      field('name', $.identifier),
      optional(field('type', $._type)),
    )),
```

It would be somewhat easy to group parameters that "reuse" the type, but I feel like that isn't the correct approach.

I'll come back to this later this week, but if anyone has additional ideas, I'm all ears 😁

Reply:
#119

Reply:
@amaanq thanks for fixing it, I was definitely over complicating this lol